lib/internal/dns/compat.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+const {
+  ArrayPrototypeFilter,
+  ArrayPrototypeMap,
+  ArrayPrototypeSort,
+  JSONParse,
+} = primordials;
+
+const { DNSWrap } = internalBinding('dns');
+const { dnsException } = require('internal/errors');
+const { AsyncResource } = require('async_hooks');
+
+class ReqWrap {
+  #resource;
+  #oncomplete;
+  constructor(name) {
+    this.#resource = new AsyncResource(name);
+  }
+
+  set oncomplete(f) {
+    this.#oncomplete = function(...args) {
+      try {
+        return this.#resource.runInAsyncScope(f, this, ...args);
+      } finally {
+        this.#resource.emitDestroy();
+      }
+    };
+  }
+
+  get oncomplete() {
+    return this.#oncomplete;
+  }
+}
+
+class QueryReqWrap extends ReqWrap {
+  constructor() {
+    super('QUERYREQWRAP');
+  }
+}
+
+class GetAddrInfoReqWrap extends ReqWrap {
+  constructor() {
+    super('GETADDRINFOREQWRAP');
+  }
+}
+
+class GetNameInfoReqWrap extends ReqWrap {
+  constructor() {
+    super('GETNAMEINFOREQWRAP');
+  }
+}
+
+[
+  ['Any', 255],
+  ['A', 1],
+  ['Aaaa', 28],
+  ['Caa', 257],
+  ['Cname', 5],
+  ['Mx', 15],
+  ['Ns', 2],
+  ['Txt', 16],
+  ['Srv', 33],
+  ['Ptr', 12],
+  ['Naptr', 35],
+  ['Soa', 6],
```

Comment:
It would be good to add a comment in here that at least indicates where these numbers come from